src/e2e/java/uk/gov/hmcts/reform/sscs/functional/sya/notifications/DocmosisWithGovNotifyLetterFunctionalTest.java

```diff
@@ -1,40 +1,91 @@
 package uk.gov.hmcts.reform.sscs.functional.sya.notifications;
 
-import static org.junit.Assert.assertEquals;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatNoException;
+import static uk.gov.hmcts.reform.sscs.config.NotificationEventTypeLists.DOCMOSIS_LETTERS;
 import static uk.gov.hmcts.reform.sscs.domain.notify.NotificationEventType.*;
 
-import java.io.IOException;
+import java.util.EnumSet;
+import java.util.HashSet;
 import java.util.List;
+import java.util.Set;
 import junitparams.JUnitParamsRunner;
 import junitparams.Parameters;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import uk.gov.hmcts.reform.sscs.domain.notify.NotificationEventType;
 import uk.gov.hmcts.reform.sscs.functional.AbstractFunctionalTest;
 import uk.gov.service.notify.Notification;
-import uk.gov.service.notify.NotificationClientException;
 
 @RunWith(JUnitParamsRunner.class)
 public class DocmosisWithGovNotifyLetterFunctionalTest extends AbstractFunctionalTest {
 
+    public static final String EXPECTED_LETTER_SUBJECT = "Pre-compiled PDF";
+    //TODO: Add callback jsons for these letters to test them functionally
+    private static final Set<NotificationEventType> DOCMOSIS_LETTERS_WITH_NO_TEST_CALLBACK = EnumSet.of(
+        ACTION_HEARING_RECORDING_REQUEST,
+        ACTION_POSTPONEMENT_REQUEST_WELSH,
+        ADMIN_APPEAL_WITHDRAWN,
+        APPEAL_LAPSED_NOTIFICATION,
+        APPEAL_WITHDRAWN_NOTIFICATION,
+        DECISION_ISSUED_WELSH,
+        DIRECTION_ISSUED_WELSH,
+        DWP_APPEAL_LAPSED_NOTIFICATION,
+        DWP_UPLOAD_RESPONSE_NOTIFICATION,
+        HMCTS_APPEAL_LAPSED_NOTIFICATION,
+        JOINT_PARTY_ADDED,
+        POSTPONEMENT_NOTIFICATION,
+        RESEND_APPEAL_CREATED_NOTIFICATION,
+        VALID_APPEAL_CREATED
+    );
+
     public DocmosisWithGovNotifyLetterFunctionalTest() {
         super(30);
     }
 
     @Test
     @Parameters(method = "eventTypes")
-    public void shouldSendDocmosisLetters(NotificationEventType notificationEventType, int expectedNumberOfLetters) throws IOException, NotificationClientException {
+    public void shouldSendDocmosisLetters(NotificationEventType notificationEventType) {
+        simulateCcdCallback(notificationEventType);
+
+        assertThatNoException().isThrownBy(() -> {
```

Comment:
Minor:

I liked it better with the `throws` declaration in this case.

The `assertThatNoException` is very useful, but not in this case. It's handy when you want to have multiple checks within the same test, especially if you want to check something AFTER the exception has been thrown (like some recording has been written for auditing purposes or things like that).

In this specific case it has no benefit: we really want to check that the letter subjects are correct. An exception being thrown is merely some other kind of problem that we didn't expect, not an explicit assertion to be checked.

Reply:
Removed